pkg/services/correlations/database.go

```diff
@@ -225,17 +225,42 @@ func (s CorrelationsService) getCorrelationsBySourceUID(ctx context.Context, cmd
 	return correlations, nil
 }
 
-func (s CorrelationsService) getCorrelations(ctx context.Context, cmd GetCorrelationsQuery) ([]Correlation, error) {
-	correlations := make([]Correlation, 0)
+func (s CorrelationsService) getCorrelations(ctx context.Context, cmd GetCorrelationsQuery) (GetCorrelationsResponseBody, error) {
+	result := GetCorrelationsResponseBody{
+		Correlations: make([]Correlation, 0),
+		Page:         cmd.Page,
+		Limit:        cmd.Limit,
+	}
 
 	err := s.SQLStore.WithDbSession(ctx, func(session *db.Session) error {
-		return session.Select("correlation.*").Join("", "data_source AS dss", "correlation.source_uid = dss.uid and dss.org_id = ?", cmd.OrgId).Join("", "data_source AS dst", "correlation.target_uid = dst.uid and dst.org_id = ?", cmd.OrgId).Find(&correlations)
+		offset := cmd.Limit * (cmd.Page - 1)
+
+		q := session.Select("correlation.*").Join("", "data_source AS dss", "correlation.source_uid = dss.uid and dss.org_id = ?", cmd.OrgId).Join("", "data_source AS dst", "correlation.target_uid = dst.uid and dst.org_id = ?", cmd.OrgId)
+
+		if len(cmd.SourceUIDs) > 0 {
+			q.In("dss.uid", cmd.SourceUIDs)
+		}
```

Comment:
we don't need this (and related changes) until we decide to implement filtering in the settings page, do we?

Reply:
We do need it. We added the ability for explore to get correlations filtered by source UID here https://github.com/grafana/grafana/pull/65241/files#diff-41f602993af9972aa8e1fbabbb0cea49eb5c8decaa46646d4ce44afb77840f5eR78 and use that for populating the correlations on the pane state.

Reply:
why aren't we using this instead? https://grafana.com/docs/grafana/latest/developers/http_api/correlations/#get-all-correlations-originating-from-a-given-data-source

Reply:
From what I remembered with our discussion, for mixed datasources we wanted to make one call and have the 100 limit be per call, not per datasource.

Having a filter on getting all correlations lets us only make one call, even in mixed environments, and enforce a pane-wide limit of only displaying 100 correlations.

Reply:
aaaah that's true, i forgot about mixed, my bad. 🙇

Reply:
@Elfo404 my fault for having this ticket take approximately 50000 years to complete